### ISSUE TITLE
Make toolchain more modular

### ISSUE THREAD

**Issue by dromer** [plugdata-team/plugdata-heavy-toolchain#16]
We could perhaps separate by compiler. Main thing is to be able to reduce duplication of required libraries and binaries. And make sure that everything is correctly in the path.

And then also need to be able to create the compile forms for each of the plug-able compiler toolchains.

The available frameworks could then maybe be grouped per toolchain.

I'm also thinking that docker solutions like the current MOD-devices builds have could be useful.
Bit bigger and requires more expert setup. But could be _much_ easier route for this kind of stuff in the future.

They have various build targets with each different build configs and compilers. replicating that on our own would be quite murder.

But, I'm also for figuring out a way to integrate with cloud compilers that vendors may want to offer. OWL already has their own build infra (based on outdated py2 branch of hvcc, but I have py3 stuff working already correctly). And MOD provides this for their max/gen~ (and possibly future max/rnbo~) cloud builds.

Anyway, thoughts/ideas about this anyone? :)

**Comment by timothyschoen** [comment]
Yes, I agree that would be a good step! So are you suggesting we move to an online service for compilation? I think it might be a good idea, and definitely less prone to errors, but unfortunately I have no experience with setting up complicated web services like that, so I'll need some help

**Comment by dromer** [comment]
No I don't want to host my own cloud infra. Really prefer the local approach!

But cloud-integration as an option to extend the program. Why not?

Either way a more modular way of specifying a toolchain module capabilities and its requirements. And allow for easier experimentation with alternate extensions for this integration.

Hmm, maybe we can do lua scripting around heavy tooling? ^_^
